### PR TITLE
Data persistence

### PR DESCRIPTION
**Pull request by jfking**

### CODE REVIEW

**Comment by pelmers** [comment]
@Vegetable26 It looks like your changes in 2345321 use spaces rather than tabs.

**Comment by pelmers** [comment]
some bookkeeping, this will fix #46.

File: native-filesystem/include/native_filesystem.h [hunk 1]
```diff
@@ -5,22 +5,103 @@
...
 		std::string getBlock(uint64_t, bool&);
+		/**
+		 * Delete contents of provided block id from this datanode. Return true
+		 * if delete successful, false otherwise (block id not found).
+		 */
 		bool rmBlock(uint64_t);
+		/**
+		 * Return the total capacity of this datanode.
+		 */
+		long getTotalSpace();
```
Comment (efrederic) [inline native-filesystem/include/native_filesystem.h]: can we make these return uint64_t?

File: native-filesystem/source/native_filesystem.cc [hunk 2]
```diff
@@ -1,115 +1,289 @@
...
-			return "";
+	bool NativeFS::writeBlock(uint64_t id, std::string blk) {
+		size_t len = blk.size();
+		uint64_t offset;
+		{
+			std::lock_guard<std::mutex> lock(listMtx);
+			if (!allocateBlock(len, offset)) {
+				LOG(ERROR) << CLASS_NAME << "Could not find a free block to fit " << len;
+			    return false;
+            }
```
Comment (efrederic) [inline native-filesystem/source/native_filesystem.cc]: spacing got a bit out of whack here

File: native-filesystem/include/native_filesystem.h [hunk 3]
```diff
@@ -5,22 +5,97 @@
...
+		NativeFS(std::string);
+		/**
+		 * Destroy NativeFS, free block array.
+		 */
+		~NativeFS();
+		/**
+		 * Write provided block contents to a block with given id. On success,
+		 * return true. Otherwise false (e.g. no free space).
+		 */
 		bool writeBlock(uint64_t, std::string);
```
Comment (nichhk) [inline native-filesystem/include/native_filesystem.h]: string ref

File: native-filesystem/include/native_filesystem.h [hunk 4]
```diff
@@ -5,22 +5,97 @@
...
+		/**
+		 * Get the contents of the block with given id. If successful, return
+		 * the contents of the block and set provided bool to true. On failure
+		 * to find the block, return empty string and set bool to false.
+		 */
 		std::string getBlock(uint64_t, bool&);
```
Comment (nichhk) [inline native-filesystem/include/native_filesystem.h]: strange that this returns a string when the rest return a bool

**Review by nichhk — CHANGES_REQUESTED** [review]

**Review by nichhk — APPROVED** [review]

**Pull request merged**